cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/TimePartitionedFileSetDataset.java

```diff
@@ -101,11 +103,26 @@ public TimePartitionedFileSetDataset(DatasetContext datasetContext, String name,
 
   @Override
   public void addPartition(long time, String path) {
+    addPartition(time, path, ImmutableMap.<String, String>of());
```

Comment:
You can use Collections.emptyMap(), instead of creating a new immutable map every time it is called